<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>612</width>
    <height>496</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Fraktal Explorer - In memories of Benoît Mandelbrot (* 20.11.1924; † 14.10.2010)</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>:/Logo/Logo.png</normaloff>:/Logo/Logo.png</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <property name="leftMargin">
     <number>1</number>
    </property>
    <property name="topMargin">
     <number>1</number>
    </property>
    <property name="rightMargin">
     <number>1</number>
    </property>
    <property name="bottomMargin">
     <number>1</number>
    </property>
    <item>
     <widget class="FraktalWidget" name="widget" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>612</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionSave_Picture"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <widget class="QMenu" name="menuDocks">
     <property name="title">
      <string>Docks</string>
     </property>
     <addaction name="separator"/>
    </widget>
    <widget class="QMenu" name="menuLanguage">
     <property name="title">
      <string>Language</string>
     </property>
     <addaction name="separator"/>
    </widget>
    <widget class="QMenu" name="menuToolbars">
     <property name="title">
      <string>Toolbars</string>
     </property>
     <addaction name="separator"/>
    </widget>
    <addaction name="actionFullscreen"/>
    <addaction name="separator"/>
    <addaction name="menuLanguage"/>
    <addaction name="separator"/>
    <addaction name="menuToolbars"/>
    <addaction name="menuDocks"/>
   </widget>
   <widget class="QMenu" name="menuAlgorithm">
    <property name="title">
     <string>Algorithm</string>
    </property>
    <addaction name="actionDetails"/>
    <addaction name="separator"/>
    <addaction name="actionNew_algorithm"/>
    <addaction name="separator"/>
    <addaction name="actionDatabank"/>
    <addaction name="actionSave_databank"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionHelp"/>
    <addaction name="separator"/>
    <addaction name="actionAbout_Fraktal_Explorer"/>
    <addaction name="actionAbout_Qt"/>
   </widget>
   <widget class="QMenu" name="menuRendering">
    <property name="title">
     <string>Rendering</string>
    </property>
    <addaction name="actionStop"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuView"/>
   <addaction name="menuRendering"/>
   <addaction name="menuAlgorithm"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QDockWidget" name="dockWidget">
   <property name="windowTitle">
    <string>Properties</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QGridLayout" name="gridLayout_2">
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item row="0" column="0">
      <widget class="QLabel" name="label_5">
       <property name="text">
        <string>Fraktal:</string>
       </property>
       <property name="buddy">
        <cstring>comboBox</cstring>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QComboBox" name="comboBox">
       <property name="toolTip">
        <string>Choose a fraktal to see him</string>
       </property>
       <property name="maxVisibleItems">
        <number>20</number>
       </property>
      </widget>
     </item>
     <item row="1" column="0" colspan="2">
      <widget class="Line" name="line">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Center X:</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QLabel" name="label_7">
       <property name="toolTip">
        <string>The current x position of the center</string>
       </property>
       <property name="text">
        <string>0,0</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>Center Y:</string>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QLabel" name="label_8">
       <property name="toolTip">
        <string>The current y position of the center</string>
       </property>
       <property name="text">
        <string>0,0</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
     <item row="5" column="0" colspan="2">
      <widget class="Line" name="line_2">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="6" column="0">
      <widget class="QLabel" name="label_4">
       <property name="text">
        <string>Zoom Faktor:</string>
       </property>
       <property name="buddy">
        <cstring>doubleSpinBox_4</cstring>
       </property>
      </widget>
     </item>
     <item row="6" column="1">
      <widget class="QDoubleSpinBox" name="doubleSpinBox_4">
       <property name="toolTip">
        <string>The used zoom faktor</string>
       </property>
       <property name="buttonSymbols">
        <enum>QAbstractSpinBox::PlusMinus</enum>
       </property>
       <property name="minimum">
        <double>0.010000000000000</double>
       </property>
       <property name="maximum">
        <double>20.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.100000000000000</double>
       </property>
       <property name="value">
        <double>0.800000000000000</double>
       </property>
      </widget>
     </item>
     <item row="7" column="0" colspan="2">
      <widget class="QCheckBox" name="checkBox">
       <property name="toolTip">
        <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'MS Shell Dlg 2'; font-size:8.25pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;Enable antialiasing for better pictures&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;(&lt;/span&gt;&lt;span style=&quot; font-size:8pt; font-weight:600;&quot;&gt;Attention&lt;/span&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;: Only useable by algorithm who are based on the L-System)&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="text">
        <string>Antialiasing</string>
       </property>
      </widget>
     </item>
     <item row="8" column="0" colspan="2">
      <widget class="Line" name="line_3">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="9" column="0">
      <widget class="QLabel" name="label_14">
       <property name="text">
        <string>Max Iterations:</string>
       </property>
       <property name="buddy">
        <cstring>spinBox_5</cstring>
       </property>
      </widget>
     </item>
     <item row="9" column="1">
      <widget class="QSpinBox" name="spinBox_5">
       <property name="toolTip">
        <string>The used max iteration level</string>
       </property>
       <property name="buttonSymbols">
        <enum>QAbstractSpinBox::PlusMinus</enum>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>1000000</number>
       </property>
       <property name="value">
        <number>64</number>
       </property>
      </widget>
     </item>
     <item row="10" column="0">
      <widget class="QLabel" name="label_15">
       <property name="text">
        <string>Limit:</string>
       </property>
       <property name="buddy">
        <cstring>spinBox_6</cstring>
       </property>
      </widget>
     </item>
     <item row="10" column="1">
      <widget class="QSpinBox" name="spinBox_6">
       <property name="toolTip">
        <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'MS Shell Dlg 2'; font-size:8.25pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;The used function deep or limitation of the algorithm result&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;(Based on the use of L-System or algebra)&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="buttonSymbols">
        <enum>QAbstractSpinBox::PlusMinus</enum>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>1000000000</number>
       </property>
       <property name="value">
        <number>4</number>
       </property>
      </widget>
     </item>
     <item row="11" column="0" colspan="2">
      <widget class="Line" name="line_4">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="4" column="0">
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>Scale:</string>
       </property>
      </widget>
     </item>
     <item row="4" column="1">
      <widget class="QLabel" name="label_6">
       <property name="toolTip">
        <string>The current scale</string>
       </property>
       <property name="text">
        <string>0,0</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
     <item row="12" column="0">
      <widget class="QLabel" name="label_9">
       <property name="text">
        <string>Length:</string>
       </property>
       <property name="buddy">
        <cstring>doubleSpinBox</cstring>
       </property>
      </widget>
     </item>
     <item row="13" column="0">
      <widget class="QLabel" name="label_10">
       <property name="text">
        <string>Delta:</string>
       </property>
       <property name="buddy">
        <cstring>spinBox</cstring>
       </property>
      </widget>
     </item>
     <item row="12" column="1">
      <widget class="QDoubleSpinBox" name="doubleSpinBox">
       <property name="toolTip">
        <string>The used step length in the L-System mode</string>
       </property>
       <property name="buttonSymbols">
        <enum>QAbstractSpinBox::PlusMinus</enum>
       </property>
       <property name="decimals">
        <number>6</number>
       </property>
       <property name="minimum">
        <double>0.000001000000000</double>
       </property>
       <property name="maximum">
        <double>1000000.000000000000000</double>
       </property>
       <property name="value">
        <double>1.000000000000000</double>
       </property>
      </widget>
     </item>
     <item row="13" column="1">
      <widget class="QSpinBox" name="spinBox">
       <property name="toolTip">
        <string>The used delta in L-System mode</string>
       </property>
       <property name="buttonSymbols">
        <enum>QAbstractSpinBox::PlusMinus</enum>
       </property>
       <property name="maximum">
        <number>360</number>
       </property>
       <property name="value">
        <number>30</number>
       </property>
      </widget>
     </item>
     <item row="14" column="0" colspan="2">
      <widget class="Line" name="line_5">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="15" column="0" colspan="2">
      <spacer name="verticalSpacer">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>File and render toolbar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionSave_Picture"/>
   <addaction name="separator"/>
   <addaction name="actionStop"/>
  </widget>
  <widget class="QToolBar" name="toolBar_2">
   <property name="windowTitle">
    <string>Algorithm toolbar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionNew_algorithm"/>
   <addaction name="actionDetails"/>
   <addaction name="separator"/>
   <addaction name="actionDatabank"/>
  </widget>
  <action name="actionQuit">
   <property name="icon">
    <iconset resource="Images.qrc">
     <normaloff>:/Icons/images/icons/Close.png</normaloff>:/Icons/images/icons/Close.png</iconset>
   </property>
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionFullscreen">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="Images.qrc">
     <normaloff>:/Icons/images/icons/Fullscreen.png</normaloff>:/Icons/images/icons/Fullscreen.png</iconset>
   </property>
   <property name="text">
    <string>Fullscreen</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+F</string>
   </property>
  </action>
  <action name="actionSave_Picture">
   <property name="icon">
    <iconset resource="Images.qrc">
     <normaloff>:/Icons/images/icons/SavePicture.png</normaloff>:/Icons/images/icons/SavePicture.png</iconset>
   </property>
   <property name="text">
    <string>Save Picture</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionDetails">
   <property name="text">
    <string>Details</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+D</string>
   </property>
  </action>
  <action name="actionNew_algorithm">
   <property name="text">
    <string>New algorithm</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionHelp">
   <property name="text">
    <string>Help</string>
   </property>
   <property name="shortcut">
    <string>F1</string>
   </property>
  </action>
  <action name="actionAbout_Fraktal_Explorer">
   <property name="text">
    <string>About Fraktal Explorer</string>
   </property>
   <property name="shortcut">
    <string>F12</string>
   </property>
  </action>
  <action name="actionAbout_Qt">
   <property name="text">
    <string>About Qt</string>
   </property>
   <property name="shortcut">
    <string>F11</string>
   </property>
   <property name="menuRole">
    <enum>QAction::AboutQtRole</enum>
   </property>
  </action>
  <action name="actionDatabank">
   <property name="text">
    <string>Databank</string>
   </property>
   <property name="shortcut">
    <string>F2</string>
   </property>
  </action>
  <action name="actionStop">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="Images.qrc">
     <normaloff>:/Icons/images/icons/Stop.png</normaloff>:/Icons/images/icons/Stop.png</iconset>
   </property>
   <property name="text">
    <string>Stop</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Space</string>
   </property>
  </action>
  <action name="actionNew_note">
   <property name="icon">
    <iconset resource="Images.qrc">
     <normaloff>:/Icons/images/icons/Note.png</normaloff>:/Icons/images/icons/Note.png</iconset>
   </property>
   <property name="text">
    <string>New note</string>
   </property>
  </action>
  <action name="actionSave_databank">
   <property name="text">
    <string>Save databank</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>FraktalWidget</class>
   <extends>QWidget</extends>
   <header>fraktalwidget.h</header>
   <container>1</container>
   <slots>
    <signal>centerXChanged(double)</signal>
    <signal>centerYChanged(double)</signal>
    <signal>scaleChanged(double)</signal>
    <slot>setCenterX(double)</slot>
    <slot>setCenterY(double)</slot>
    <slot>setScale(double)</slot>
    <slot>setZoomFaktor(double)</slot>
    <slot>setFraktal(int)</slot>
    <slot>setAntialiasing(bool)</slot>
    <slot>setMaxIterations(int)</slot>
    <slot>setLimit(int)</slot>
    <slot>stopRendering()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="Images.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionQuit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>doubleSpinBox_4</sender>
   <signal>valueChanged(double)</signal>
   <receiver>widget</receiver>
   <slot>setZoomFaktor(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>591</x>
     <y>161</y>
    </hint>
    <hint type="destinationlabel">
     <x>140</x>
     <y>191</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>widget</sender>
   <signal>scaleChanged(double)</signal>
   <receiver>label_6</receiver>
   <slot>setNum(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>218</x>
     <y>150</y>
    </hint>
    <hint type="destinationlabel">
     <x>602</x>
     <y>137</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>widget</sender>
   <signal>centerXChanged(double)</signal>
   <receiver>label_7</receiver>
   <slot>setNum(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>226</x>
     <y>97</y>
    </hint>
    <hint type="destinationlabel">
     <x>602</x>
     <y>99</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>widget</sender>
   <signal>centerYChanged(double)</signal>
   <receiver>label_8</receiver>
   <slot>setNum(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>221</x>
     <y>120</y>
    </hint>
    <hint type="destinationlabel">
     <x>602</x>
     <y>118</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>widget</receiver>
   <slot>setAntialiasing(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>591</x>
     <y>184</y>
    </hint>
    <hint type="destinationlabel">
     <x>265</x>
     <y>386</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox_6</sender>
   <signal>valueChanged(int)</signal>
   <receiver>widget</receiver>
   <slot>setLimit(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>602</x>
     <y>256</y>
    </hint>
    <hint type="destinationlabel">
     <x>263</x>
     <y>430</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox_5</sender>
   <signal>valueChanged(int)</signal>
   <receiver>widget</receiver>
   <slot>setMaxIterations(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>602</x>
     <y>230</y>
    </hint>
    <hint type="destinationlabel">
     <x>242</x>
     <y>396</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionStop</sender>
   <signal>triggered()</signal>
   <receiver>widget</receiver>
   <slot>stopRendering()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>204</x>
     <y>248</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
